pkg/controller/bastion/options.go

```diff
@@ -0,0 +1,137 @@
+// Copyright (c) 2021 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package bastion
+
+import (
+	"crypto/sha256"
+	"encoding/base64"
+	"fmt"
+	"net"
+
+	"github.com/gardener/gardener/extensions/pkg/controller"
+	v1beta1constants "github.com/gardener/gardener/pkg/apis/core/v1beta1/constants"
+	extensionsv1alpha1 "github.com/gardener/gardener/pkg/apis/extensions/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+)
+
+//Maximum length for "base" name due to fact that we use this name to name other Alicloud resources,
+const (
+	maxLengthForBaseName = 33
+	imageID              = "aliyun_3_x64_20G_scc_alibase_20211018.vhd"
+	instanceType         = "ecs.g5.large"
```

Comment:
Available instance types vary from zones to zones. We'd better calculate the instance type from a certain rules (for example, minimum cpu & memory).

Reply:
committed per comment 3e388e97  @EmoinLanyu